UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java

```diff
@@ -288,8 +320,77 @@ public void brakeBtn(View view) {
         brake();
     }
 
+    /**
+     * Helper method that simulates the database retrieving a bus line. We assume the drivers have assigned bus lines by the system.
+     */
+    private void generateBusLine(){
+        busLine = new BusLine("9");
+        busLine.addStop("Sandarna");
+        busLine.addStop("Sannaplan");
+        busLine.addStop("Mariaplan");
+        busLine.addStop("Vagnhallen Majorna");
+
+        busLineName.setText("Line: " + busLine.getName());
+    }
+
+    /**
+     * Change the next stop on the driver's screen and publish it to communicate with the passengers.
+     * If the end of the line, display an information to the driver and invite to click one more time to reverse the line.
+     * @param view -> "next" button (R.id.changeStop)
+     */
+    public void changeStop(View view){
+        String comingStop = busLine.nextStop();
+
+        if (comingStop.equals(BusLine.NO_NEXT_STOP)){
+            Toast reverseInstruction = Toast.makeText(this, BusLine.REVERSE_INSTRUCTION, Toast.LENGTH_LONG);
+            reverseInstruction.setGravity(TOP, 0,0);
+            reverseInstruction.show();
+        }
+            publishNextStop(comingStop);
+            displayNextStop(comingStop);
+    }
+
+    /**
+     * Helper method when pressing the next stop button.
+     * Send the next stop via MQTT to the topic "NEXT_STOP + name of bus line" (In case we have several lines).
+     * The passengers should subscribe to this topic when choosing the line they are using.
+     * @param comingStop -> the next stop as a string
+     */
+    private void publishNextStop(String comingStop){
```

Comment:
We should store this information as a shared preference as opposed to sending via MQTT, since it is data that is only client-facing. Unless the data is being passed onto the car itself, we should avoid message transport via MQTT as we already have a tremendous backlog that is increasing our latency.

Reply:
The problem if we use the shared preference, is that we are not sending data from one user to the other but from one profile to the other on the same device. Which is not what it is supposed to do. 
This joins the conversation on the #85, I would like more insights from the rest of the group regarding that. 

On the other hand, this is for a user story that is not implemented and is low priority #75. We should have a meeting and decide if it is worth implementing it. And if so, if we use MQTT or shared preference. 

I understand the latency issue, I am just not sure what to decide about the shared preference solution. And I am not able to test the MQTT because of the camera feed which is killing my computer.

Reply:
Pretend you're a bored passenger using the Pathfinder app, and you wanna do something else while you wait. You click out of the app and go on Spotify, Facebook, whatever. Shared preference saves your last input, and loads your last input from when you last ran the app.

So, if you pressed the stop button, and somehow left the app, when you come back it will still have the Stop button toggled on. Without it, even if you saved the message into MQTT, the sign may flash but the stop and accessibility buttons would default back to off. This would be very confusing to the user. 
If you have tinkered with the passenger_screen branch, you'd see that the "Stop" button changes text to "Cancel Stop" and vice versa when toggled.

An idea that comes to mind would be to publish an MQTT message in a shared preference function. That way all necessary data gets stored and you get to pass data along from one process to another.